instrumentation/github.com/gin-gonic/gin/otelgin/option.go

```diff
@@ -91,3 +99,19 @@ func WithSpanNameFormatter(f func(r *http.Request) string) Option {
 		c.SpanNameFormatter = f
 	})
 }
+
+// WithMeterProvider specifies a meter provider to use for creating a meter.
+// If none is specified, the global provider is used.
+func WithMeterProvider(mp metric.MeterProvider) Option {
+	return optionFunc(func(c *config) {
+		c.MeterProvider = mp
+	})
+}
+
+// WithMetricAttributeExtractor specifies a function that extracts additional attributes from the http.Request
+// and returns them as a slice of attribute.KeyValue.
+func WithMetricAttributeExtractor(f MetricAttributeExtractor) Option {
```

Comment:
Could we use the same pattern/API as in otelhttp here, with [WithMetricAttributesFn](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#WithMetricAttributesFn)?